### PR TITLE
Refactoring for shogun::memcpy

### PR DESCRIPTION
**Pull request by MikeLing**

### CODE REVIEW

**Comment by MikeLing** [comment]
#3700

**Comment by MikeLing** [comment]
@lambday  ask review for this pr, thank you;)

File: src/shogun/base/Parameter.cpp [hunk 1]
```diff
@@ -3354,7 +3354,7 @@ bool TParameter::copy_ptype(EPrimitiveType ptype, void* source, void* target)
 {
 	SG_SDEBUG("entering TParameter::copy_ptype()\n");
 
-	/* rather than using memcpy, use the cumbersome way here and cast all types.
+	/* rather than using sg_memcpy, use the cumbersome way here and cast all types.
```
Comment (lambday) [inline src/shogun/base/Parameter.cpp]: haha `sed` all over :D.. love it ;)

**Comment by lambday** [comment]
@MikeLing looks good to me.. merging this one..

But can you check whether we can use it in swig typemaps (Just `grep -Rin "memcpy" src/interfaces`).

**Comment by MikeLing** [comment]
@lambday yep, we are using it in swig. Do we need to address it in swig, too?

**Pull request merged**